inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/ClassInstrumentationConfiguration.java

```diff
@@ -0,0 +1,70 @@
+package rocks.inspectit.oce.core.instrumentation;
+
+import lombok.Value;
+import net.bytebuddy.description.type.TypeDescription;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.instrumentation.special.SpecialSensor;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Defines the actual or requested state of the instrumentation of a single class.
+ */
+@Value
+class ClassInstrumentationConfiguration {
+
+    /**
+     * The configuration representing that no instrumentation of the class is performed.
+     */
+    static final ClassInstrumentationConfiguration NO_INSTRUMENTATION = new ClassInstrumentationConfiguration(Collections.emptySet(), null);
+
+    /**
+     * The sensors which are active for the target class.
+     */
+    Set<SpecialSensor> activeSensors;
+
+    /**
+     * The inspectIT configuration used to derive this configuration object.
+     * Can be null, but only if no sensors active.
+     */
+    InstrumentationSettings activeConfiguration;
```

Comment:
let's make all the fields `final` and `private`?

Reply:
Done by lombok @Value.